sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -164,9 +165,12 @@ case class BucketSpec(
     numBuckets: Int,
     bucketColumnNames: Seq[String],
     sortColumnNames: Seq[String]) {
-  if (numBuckets <= 0 || numBuckets >= 100000) {
+  def conf: SQLConf = SQLConf.get
+
+  if (numBuckets <= 0 || numBuckets > conf.bucketingMaxBuckets) {
```

Comment:
Since the condition is changed from `>` to `>=`, there is inconsistent between the condition and the error message.

If this condition is true, the message is like `... but less than or equal to bucketing.maxBuckets ...`.

Reply:
Could you submit a followup PR to address this message issue?

Reply:
We can merge this PR first.